app/src/main/java/com/kickstarter/viewmodels/ActivityFeed.java

```diff
@@ -0,0 +1,252 @@
+package com.kickstarter.viewmodels;
+
+import android.support.annotation.NonNull;
+
+import com.kickstarter.libs.ActivityViewModel;
+import com.kickstarter.libs.ApiPaginator;
+import com.kickstarter.libs.CurrentUserType;
+import com.kickstarter.libs.Environment;
+import com.kickstarter.libs.rx.transformers.Transformers;
+import com.kickstarter.models.Activity;
+import com.kickstarter.models.Project;
+import com.kickstarter.services.ApiClientType;
+import com.kickstarter.services.apiresponses.ActivityEnvelope;
+import com.kickstarter.ui.activities.ActivityFeedActivity;
+import com.kickstarter.ui.adapters.ActivityFeedAdapter;
+import com.kickstarter.ui.viewholders.EmptyActivityFeedViewHolder;
+import com.kickstarter.ui.viewholders.FriendBackingViewHolder;
+import com.kickstarter.ui.viewholders.ProjectStateChangedPositiveViewHolder;
+import com.kickstarter.ui.viewholders.ProjectStateChangedViewHolder;
+import com.kickstarter.ui.viewholders.ProjectUpdateViewHolder;
+
+import java.util.List;
+
+import rx.Observable;
+import rx.subjects.BehaviorSubject;
+import rx.subjects.PublishSubject;
+
+public interface ActivityFeed {
+
+  interface Inputs extends ActivityFeedAdapter.Delegate {
+    /**
+     * Invoke when pagination should happen.
+     */
```

Comment:
maybe we should single-line these comments like so:
```java
/** Invoke when pagination should happen. */
```
that way we can keep this comment style if need be for longer comments, but keep these interfaces from getting too long